sockeye/rerank.py

```diff
@@ -90,6 +92,7 @@ def rerank(args: argparse.Namespace):
     :param args: Namespace object holding CLI arguments.
     """
     reranker = Reranker(args.metric, args.return_score)
+    output_stream = sys.stdout if args.output is None else smart_open(args.output, mode='w')
```

Comment:
Can we move the open into a with statement or use a context manager to ensure we close the stream?

Reply:
Good catch.  For this PR, I added a low-tech solution of just closing the stream when it's not `sys.stdout`.

I like the idea of a context manager for "file or stdout", probably as part of the utils module with proper unit tests.  Right now low-tech is less code overall, but if we find ourselves using this pattern again, we can add the utility function.